File: ruby/organizing_db_adapter/lib/organizing/in_memory_game_repository.rb
require 'organizing/game_repository'

module Organizing
  class InMemoryGameRepository
    include GameRepository

    def initialize
      @store = {}
    end

    def save(game)
      @store[game.id] = game
    end

    def find_by_id(game_id)
      @store[game_id]
    end

    def find_all
      @store.values
    end
  end
end

File: java/organizing-db-adapter/src/main/java/com/hexagonalchess/organizing/InMemoryGameRepository.java
package com.hexagonalchess.organizing;

import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;
import java.util.Optional;

public class InMemoryGameRepository implements GameRepository {

    private final Map<GameId, Game> store = new LinkedHashMap<>();

    @Override
    public void save(Game game) {
        store.put(game.id(), game);
    }

    @Override
    public Optional<Game> findById(GameId gameId) {
        return Optional.ofNullable(store.get(gameId));
    }

    @Override
    public List<Game> findAll() {
        return new ArrayList<>(store.values());
    }
}
